backend/app/models/game.rb

```diff
@@ -0,0 +1,123 @@
+class Game < ApplicationRecord
+  belongs_to :player1, class_name: "User"
+  belongs_to :player2, class_name: "User", optional: true
+  has_many :moves, dependent: :destroy
+  belongs_to :winner, class_name: 'User', optional: true
+  belongs_to :last_move, class_name: 'Move', optional: true
+  include Workflow
+  workflow do
+    state :new do
+      event :start_play, :transitions_to => :playing
+    end
+    state :playing do
+      event :game_completed, :transitions_to => :game_completed
+    end
+    state :game_completed
+  end
+
+
+  def self.leave_game(player)
```

Comment:
I think this method is too broad. A better distinction between `game.leave(player)` and `Game.played_by(player).leave_all` should be made